### PR TITLE
[libc][stdbit] implement stdc_first_leading_zero (C23)

### PR DESCRIPTION
**Pull request by nickdesaulniers**

### CODE REVIEW

File: libc/src/__support/CPP/bit.h [hunk 1]
```diff
@@ -238,6 +238,13 @@ LIBC_INLINE constexpr To bit_or_static_cast(const From &from) {
   }
 }
 
+template <typename T, typename = cpp::enable_if_t<cpp::is_unsigned_v<T>>>
+[[nodiscard]] LIBC_INLINE constexpr int first_leading_zero(T value) {
+  return value == cpp::numeric_limits<T>::max()
+             ? 0
+             : countl_zero(static_cast<T>(~value)) + 1;
```
Comment (nickdesaulniers) [inline libc/src/__support/CPP/bit.h:245]: @gchatelet perhaps I should do something akin to `ADD_SPECIALIZATION`?

```c++
return value == cpp::numeric_limits<T>::max() ? 0 : BUILTIN(static_cast<T>(~value) + 1;
```

Comment (nickdesaulniers) [reply libc/src/__support/CPP/bit.h:253]: added in 87518df24b71

File: libc/src/__support/CPP/bit.h [hunk 2]
```diff
@@ -238,13 +238,36 @@ LIBC_INLINE constexpr To bit_or_static_cast(const From &from) {
   }
 }
 
+#define ADD_SPECIALIZATION(NAME, TYPE, BUILTIN)                                \
```
Comment (lntue) [inline libc/src/__support/CPP/bit.h:241]: Can we have different macro names to distinguish between this `ADD_SPECIALIZATION` and the previous `ADD_SPECIALIZATION`?  It is going to be a bit tricky for readability to see which definition apply to which `ADD_SPECIALIZATION` when searching.

Comment (nickdesaulniers) [reply libc/src/__support/CPP/bit.h]: done in cea5f1fe0cef PTAL

**Review by lntue — APPROVED** [review]

**Pull request merged**